Source/WebKit/NetworkProcess/cache/NetworkCache.cpp

```diff
@@ -381,9 +381,11 @@ void Cache::startAsyncRevalidationIfNeeded(const WebCore::ResourceRequest& reque
         auto addResult = m_pendingAsyncRevalidationByPage.ensure(frameID, [] {
             return WeakHashSet<AsyncRevalidation>();
         });
-        Ref revalidation = AsyncRevalidation::create(*this, frameID, request, WTFMove(entry), isNavigatingToAppBoundDomain, allowPrivacyProxy, advancedPrivacyProtections, [this, key](auto result) {
-            ASSERT(m_pendingAsyncRevalidations.contains(key));
-            m_pendingAsyncRevalidations.remove(key);
+        Ref revalidation = AsyncRevalidation::create(*this, frameID, request, WTFMove(entry), isNavigatingToAppBoundDomain, allowPrivacyProxy, advancedPrivacyProtections, [weakThis = WeakPtr { *this }, key](auto result) {
+            if (weakThis) {
```

Comment:
Why no `protectedThis` here? Seems a little risky to be using `weakThis` without protection.

Reply:
I assumed that because all uses of `weakThis` are trivial, it wouldn't matter, but I can change it if you prefer.

Reply:
Yes, I think we just avoid any risk in modern code, especially when there is no performance concern.